### ISSUE TITLE
Add a GitHub Actions workflow to build & publish choose in GH release ? 

### ISSUE THREAD

**Issue by rgarrigue** [theryangeary/choose#24]
Hey there

Would you mind adding a pre-build binary to your GH release ? I never tried building & publishing Rust based code before, but from a past experience using rust based tool, here's a starter.

```yaml
name: Publish binary

on: 
  push:

jobs:
  ebook:
    name: Build and upload
    runs-on: ubuntu-latest
    steps:
      # Install Rust & its package manager Cargo
      - name: Install cargo
        uses: actions-rs/toolchain@v1
        with:
          toolchain: stable
          profile: minimal

      # Cache installation assets
      - name: Cache cargo registry
        uses: actions/cache@v1
        with:
          path: ~/.cargo/registry
          key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.lock') }}
      - name: Cache cargo index
        uses: actions/cache@v1
        with:
          path: ~/.cargo/git
          key: ${{ runner.os }}-cargo-index-${{ hashFiles('**/Cargo.lock') }}
      - name: Cache cargo build
        uses: actions/cache@v1
        with:
          path: target
          key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('**/Cargo.lock') }}

      # BUILD HERE
      - name: build
        run: cargo ??? 

      # Publish
      - name: Create Release
        id: create-release
        uses: actions/create-release@v1.0.0
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          tag_name: ${{ github.ref }}
          release_name: Release ${{ github.ref }}
          draft: false
          prerelease: false

      - name: Upload to release
        id: upload-release-asset
        uses: actions/upload-release-asset@v1.0.1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          upload_url: ${{ steps.create-release.outputs.upload_url }}
          asset_path: ./PATH
          asset_name: NAME
```

Best regards,

**Comment by Slach** [comment]
yes, I think github releases will be  best choice

**Comment by theryangeary** [comment]
I agree, this is a good idea. Haven't had a chance yet but I think it is now next on my agenda.

**Comment by theryangeary** [comment]
Hi folks, happy to say that this is resolved in `abde654`.
